plugins/base/src/main/kotlin/translators/psi/JavadocParser.kt

```diff
@@ -43,6 +44,40 @@ class JavadocParser(
         return DocumentationNode(nodes)
     }
 
+    private fun findClosestDocComment(element: PsiNamedElement): PsiDocComment? {
```

Comment:
@sellophane 
This turned out to be not as simple as we might have hoped. I am here reporting a potential conflict in documentation when there is a Interface and a base class that would provide proper documentation for the same method.

Reply:
I did some testing and it seems it returns only the most immediate ancestor - so if there is a structure like:
classA <- classB <- classC <- InterfaceC1, only classB method will be returned. For this, much simpler recursive function should be sufficient. 
In what scenario you encountered yours behaviour?

Reply:
I also cover the following hierarchies:

1.

```kotlin
BaseClass {
    /** Documentation */
    fun x() 
}

InterfaceX {
    /** Other Documentation */
    fun x()
}

class X: BaseClass(), InterfaceX {
    fun x() // <-- which documentation to pick now? Interface or class?
}
```

2.

```kotlin

class BaseClass1 {
    /** Documented */
    fun x() 
}

class BaseClass2: BaseClass1() {
    fun x() // <- Not explicitly documented
}

class X : BaseClass2() {
    fun x() // <- Not explicitly documented. Implicitly by BaseClass1
}

```

Reply:
The warning, as well as the convention class > interface was discussed with @Kordyjan